sdks/python/apache_beam/runners/worker/operations.py

```diff
@@ -299,7 +299,14 @@ def _read_side_inputs(self, tags_and_types):
         if not isinstance(si, operation_specs.WorkerSideInputSource):
           raise NotImplementedError('Unknown side input type: %r' % si)
         sources.append(si.source)
-      iterator_fn = sideinputs.get_iterator_fn_for_sources(sources)
+      si_counter = opcounters.SideInputReadCounter(
+          self.counter_factory,
+          self.state_sampler,
+          declaring_step=self.operation_name,
+          # Inputs are 1-indexed, so we add 1 to i in the side input id
+          input_index=i + 1)
```

Comment:
<!--new_thread; commit:0bfc12f8986238da5b5014cfc5eea4d32e6638f7; resolved:0-->
Please see comment in other file (opcounters.py) about stability of the order here (since we use side_tag) as the key for side inputs.